### PR TITLE
An assortment of Lighting Stuff

### PR DESCRIPTION
**Pull request by Novacat**
- Makes Lighting less uniform (Baystation PR #19808)
- Refactors Lighting stuff (Baystation PR #13614)

Ported from Baystation code.

### CODE REVIEW

**Comment by zekesulastin** [comment]
https://github.com/Baystation12/Baystation12/pull/19808 since no screenshots were posted.

File: code/game/objects/items/devices/lightreplacer.dm [hunk 1]
```diff
@@ -121,48 +121,18 @@
...
-			return
-
-		else
-			U << failmsg
-			return
-	else
-		U << "There is a working [target.fitting] already inserted."
-		return
+	if(target.status == LIGHT_OK)
+		U << "There is a working [target.get_fitting_name()] already inserted."
```
Comment (Anewbe) [inline code/game/objects/items/devices/lightreplacer.dm]: Change this into a to_chat() please.

File: code/game/objects/items/devices/lightreplacer.dm [hunk 2]
```diff
@@ -121,48 +121,18 @@
...
+	else if(!CanUse(U))
+		U << failmsg
```
Comment (Anewbe) [inline code/game/objects/items/devices/lightreplacer.dm]: Change this into a to_chat() please.

File: code/game/objects/items/devices/lightreplacer.dm [hunk 3]
```diff
@@ -121,48 +121,18 @@
...
+	else if(Use(U))
+		U << "<span class='notice'>You replace the [target.get_fitting_name()] with the [src].</span>"
```
Comment (Anewbe) [inline code/game/objects/items/devices/lightreplacer.dm]: Change this into a to_chat() please.

File: code/modules/power/lighting.dm [hunk 4]
```diff
@@ -19,14 +27,24 @@
...
+	if(fixture)
+		fixture_type = fixture.type
+		fixture.transfer_fingerprints_to(src)
+		set_dir(fixture.dir)
+		stage = 2
+	update_icon()
+
+/obj/machinery/light_construct/update_icon()
+	switch(stage)
+		if(1) icon_state = "tube-construct-stage1"
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Please put the contents of an if/else/for check under the check itself, for readability.

File: code/modules/power/lighting.dm [hunk 5]
```diff
@@ -35,15 +53,9 @@
...
-		if(1)
-			user << "It's an empty frame."
-			return
-		if(2)
-			user << "It's wired."
-			return
-		if(3)
-			user << "The casing is closed."
-			return
+		if(1) user << "It's an empty frame."
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Change this into a to_chat() please.
Additionally, please put the contents of an if/else/for check under the check itself, for readability.

File: code/modules/power/lighting.dm [hunk 6]
```diff
@@ -19,14 +27,24 @@
...
+		fixture_type = fixture.type
+		fixture.transfer_fingerprints_to(src)
+		set_dir(fixture.dir)
+		stage = 2
+	update_icon()
+
+/obj/machinery/light_construct/update_icon()
+	switch(stage)
+		if(1) icon_state = "tube-construct-stage1"
+		if(2) icon_state = "tube-construct-stage2"
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Please put the contents of an if/else/for check under the check itself, for readability.

File: code/modules/power/lighting.dm [hunk 7]
```diff
@@ -19,14 +27,24 @@
...
+		if(3) icon_state = "tube-empty"
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Please put the contents of an if/else/for check under the check itself, for readability.

File: code/modules/power/lighting.dm [hunk 8]
```diff
@@ -35,15 +53,9 @@
...
-			user << "It's an empty frame."
-			return
-		if(2)
-			user << "It's wired."
-			return
-		if(3)
-			user << "The casing is closed."
-			return
+		if(1) user << "It's an empty frame."
+		if(2) user << "It's wired."
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Change this into a to_chat() please.
Additionally, please put the contents of an if/else/for check under the check itself, for readability.

File: code/modules/power/lighting.dm [hunk 9]
```diff
@@ -35,15 +53,9 @@
...
+		if(3) user << "The casing is closed."
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Change this into a to_chat() please.
Additionally, please put the contents of an if/else/for check under the check itself, for readability.

File: code/modules/power/lighting.dm [hunk 10]
```diff
@@ -388,34 +394,15 @@
 	// attempt to insert light
 	if(istype(W, /obj/item/weapon/light))
 		if(status != LIGHT_EMPTY)
-			user << "There is a [fitting] already inserted."
+			user << "There is a [get_fitting_name()] already inserted."
+			return
+		if(!istype(W, light_type))
+			user << "This type of light requires a [get_fitting_name()]."
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Change this into a to_chat() please.

File: code/modules/power/lighting.dm [hunk 11]
```diff
@@ -388,34 +394,15 @@
 	// attempt to insert light
 	if(istype(W, /obj/item/weapon/light))
 		if(status != LIGHT_EMPTY)
-			user << "There is a [fitting] already inserted."
+			user << "There is a [get_fitting_name()] already inserted."
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Change this into a to_chat() please.

File: code/modules/power/lighting.dm [hunk 12]
```diff
@@ -388,34 +394,15 @@
...
-				if(on && rigged)
-
-					log_admin("LOG: Rigged light explosion, last touched by [fingerprintslast]")
-					message_admins("LOG: Rigged light explosion, last touched by [fingerprintslast]")
 
-					explode()
-			else
-				user << "This type of light requires a [fitting]."
-				return
+		user << "You insert [W]."
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Change this into a to_chat() please.

File: code/modules/power/lighting.dm [hunk 13]
```diff
@@ -545,7 +514,7 @@
 	add_fingerprint(user)
 
 	if(status == LIGHT_EMPTY)
-		user << "There is no [fitting] in this light."
+		user << "There is no [get_fitting_name()] in this light."
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Change this into a to_chat() please.

File: code/modules/power/lighting.dm [hunk 14]
```diff
@@ -574,39 +543,22 @@
 			prot = 1
 
 		if(prot > 0 || (COLD_RESISTANCE in user.mutations))
-			user << "You remove the light [fitting]"
+			user << "You remove the light [get_fitting_name()]"
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Change this into a to_chat() please.

File: code/modules/power/lighting.dm [hunk 15]
```diff
@@ -574,39 +543,22 @@
 			prot = 1
 
 		if(prot > 0 || (COLD_RESISTANCE in user.mutations))
-			user << "You remove the light [fitting]"
+			user << "You remove the light [get_fitting_name()]"
 		else if(TK in user.mutations)
-			user << "You telekinetically remove the light [fitting]."
+			user << "You telekinetically remove the light [get_fitting_name()]."
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Change this into a to_chat() please.

File: code/modules/power/lighting.dm [hunk 16]
```diff
@@ -574,39 +543,22 @@
...
 
 		if(prot > 0 || (COLD_RESISTANCE in user.mutations))
-			user << "You remove the light [fitting]"
+			user << "You remove the light [get_fitting_name()]"
 		else if(TK in user.mutations)
-			user << "You telekinetically remove the light [fitting]."
+			user << "You telekinetically remove the light [get_fitting_name()]."
 		else
-			user << "You try to remove the light [fitting], but it's too hot and you don't want to burn your hand."
+			user << "You try to remove the light [get_fitting_name()], but it's too hot and you don't want to burn your hand."
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Change this into a to_chat() please.

File: code/modules/power/lighting.dm [hunk 17]
```diff
@@ -574,39 +543,22 @@
...
 			return				// if burned, don't remove the light
 	else
-		user << "You remove the light [fitting]."
+		user << "You remove the light [get_fitting_name()]."
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Change this into a to_chat() please.

File: code/modules/power/lighting.dm [hunk 18]
```diff
@@ -574,39 +543,22 @@
...
-
-	status = LIGHT_EMPTY
-	update()
+	user.put_in_active_hand(remove_bulb())	//puts it in our active hand
 
 /obj/machinery/light/flamp/attack_hand(mob/user)
 	if(lamp_shade)
 		if(status == LIGHT_EMPTY)
-			user << "There is no [fitting] in this light."
+			user << "There is no [get_fitting_name()] in this light."
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Change this into a to_chat() please.

File: code/modules/power/lighting.dm [hunk 19]
```diff
@@ -621,28 +573,11 @@
 
 /obj/machinery/light/attack_tk(mob/user)
 	if(status == LIGHT_EMPTY)
-		user << "There is no [fitting] in this light."
+		user << "There is no [get_fitting_name()] in this light."
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Change this into a to_chat() please.

File: code/modules/power/lighting.dm [hunk 20]
```diff
@@ -621,28 +573,11 @@
...
-	L.switchcount = switchcount
-	switchcount = 0
-
-	L.update()
-	L.add_fingerprint(user)
-	L.loc = loc
-
-	status = LIGHT_EMPTY
-	update()
+	user << "You telekinetically remove the light [get_fitting_name()]."
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Change this into a to_chat() please.

File: code/modules/power/lighting.dm [hunk 21]
```diff
@@ -134,9 +120,15 @@
 	icon_state = "bulb-construct-stage1"
 	anchored = 1
 	stage = 1
-	fixture_type = "bulb"
+	fixture_type = /obj/machinery/light/small
 	sheets_refunded = 1
 
+/obj/machinery/light_construct/small/update_icon()
+	switch(stage)
+		if(1) icon_state = "bulb-construct-stage1"
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Please put the contents of an if/else/for check under the check itself, for readability.

File: code/modules/power/lighting.dm [hunk 22]
```diff
@@ -134,9 +120,15 @@
...
 	anchored = 1
 	stage = 1
-	fixture_type = "bulb"
+	fixture_type = /obj/machinery/light/small
 	sheets_refunded = 1
 
+/obj/machinery/light_construct/small/update_icon()
+	switch(stage)
+		if(1) icon_state = "bulb-construct-stage1"
+		if(2) icon_state = "bulb-construct-stage2"
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Please put the contents of an if/else/for check under the check itself, for readability.

File: code/modules/power/lighting.dm [hunk 23]
```diff
@@ -134,9 +120,15 @@
...
+		if(3) icon_state = "bulb-empty"
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Please put the contents of an if/else/for check under the check itself, for readability.

File: code/modules/power/lighting.dm [hunk 24]
```diff
@@ -146,9 +138,15 @@
 	plane = OBJ_PLANE
 	layer = OBJ_LAYER
 	stage = 1
-	fixture_type = "flamp"
+	fixture_type = /obj/machinery/light/flamp
 	sheets_refunded = 2
 
+/obj/machinery/light_construct/flamp/update_icon()
+	switch(stage)
+		if(1) icon_state = "flamp-construct-stage1"
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Please put the contents of an if/else/for check under the check itself, for readability.

File: code/modules/power/lighting.dm [hunk 25]
```diff
@@ -146,9 +138,15 @@
...
 	layer = OBJ_LAYER
 	stage = 1
-	fixture_type = "flamp"
+	fixture_type = /obj/machinery/light/flamp
 	sheets_refunded = 2
 
+/obj/machinery/light_construct/flamp/update_icon()
+	switch(stage)
+		if(1) icon_state = "flamp-construct-stage1"
+		if(2) icon_state = "flamp-construct-stage2"
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Please put the contents of an if/else/for check under the check itself, for readability.

File: code/modules/power/lighting.dm [hunk 26]
```diff
@@ -146,9 +138,15 @@
...
+		if(3) icon_state = "flamp-empty"
```
Comment (Anewbe) [inline code/modules/power/lighting.dm]: Please put the contents of an if/else/for check under the check itself, for readability.

**Pull request merged**